### ISSUE TITLE
[Alpha 1.0] Sway crash when moving to scratchpad in config

### ISSUE THREAD

**Issue by nycex** [swaywm/sway#2656]
Sway crashes, if you move something directly from the config to the scratchpad. here is the command:
`for_window [instance=$scratchterm] move scratchpad`
sway version: 1.0-alpha.5-362-g5abf8322 (Sep 17 2018, branch 'master')

**Comment by Marteon** [comment]
I don't know how sway handles this internally but it sounds like the porblem is the `move` part. Might be related to #2657

**Comment by nycex** [comment]
Yeah, it is the move part,
[for_window [instance=$scratchterm] floating enable](url)
and
[for_window [instance=$scratchterm] resize set 1000 600](url)
work fine

**Comment by RyanDwyer** [comment]
I looked into this. In `view_execute_criteria`, we set focus to the view while running the criteria command, then restore focus back to whatever it was previously. When the view is mapping, prior focus is the same view. The crash occurs because we're setting the focus to the hidden scratchpad container, and `seat_set_focus_warp` rightfully assumes that the node being given focus is not a hidden scratchpad container.

A naive approach would be to make `seat_set_focus_warp` handle a hidden scratchpad container correctly, or change `view_execute_criteria` to avoid setting focus if the container has moved to the scratchpad. I think the best option is to change how commands are executed so we can run commands/criteria on a view without having to give it focus.
